testsuite/integration/basic/src/test/java/org/wildfly/test/integration/microprofile/metrics/metadata/MicroProfileMetricsHistogramMultipleInvocationsTestCase.java

```diff
@@ -0,0 +1,117 @@
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.integration.microprofile.metrics.TestApplication;
+import org.wildfly.test.integration.microprofile.metrics.metadata.resources.MicroProfileMetricsHistogramResource;
+
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.wildfly.test.integration.microprofile.metrics.MetricsHelper.getJSONMetrics;
+import static org.wildfly.test.integration.microprofile.metrics.MetricsHelper.getMetricSubValueFromJSONOutput;
+
+/**
+ * Regression test for SmallRye Metrics issue:
+ * https://github.com/smallrye/smallrye-metrics/issues/42
+ * Histogram - registry.histogram(metadata) returns IllegalArgumentException on second invocation of the endpoint.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class MicroProfileMetricsHistogramMultipleInvocationsTestCase {
+
+
+   /**
+    * Json metrics
+    */
+   private static String json;
+
+   /**
+    * ManagementClient provides management port and hostname
+    */
+   @ContainerResource
+   ManagementClient managementClient;
+
+   /**
+    * Prepare deployment
+    */
+   @Deployment
+   public static Archive<?> deploy() {
+      WebArchive war = ShrinkWrap.create(WebArchive.class, MicroProfileMetricsHistogramMultipleInvocationsTestCase.class.getSimpleName() + ".war")
+            .addClasses(TestApplication.class)
+            .addClasses(MicroProfileMetricsHistogramResource.class)
+            .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+      return war;
+   }
+
+   /**
+    * Perform 10 invocations on an endpoint. After first invocation, pre-existing histogram should be used.
+    */
+   @Test
+   public void testMultipleInvocationsOfEndpoint(@ArquillianResource URL url) throws Exception {
+      try (CloseableHttpClient client = HttpClients.createDefault()) {
+         for (int i = 1; i <= 5; i++) {
+            for (int j = 1; j <= 2; j++) { // use each number from 1 to 5 twice
+               URL endpointURL = new URL(url.toExternalForm() + "microprofile-metrics-app/histogram/hello/" + i);
+               HttpGet httpGet = new HttpGet(endpointURL.toExternalForm());
+               CloseableHttpResponse resp = client.execute(httpGet);
+               assertEquals(200, resp.getStatusLine().getStatusCode());
+               String content = EntityUtils.toString(resp.getEntity());
+               assertTrue("'Hello World!' message is expected, but was: " + content, content.contains("Hello World!"));
+            }
+         }
+      }
+
+      json = getJSONMetrics(managementClient, "application/helloHistogram", true);
+
+      checkValue("min", 1.0);
+      checkValue("max", 5.0);
+      checkValue("mean", 3.0);
+      checkValue("stddev", 1.4142135623730951);
```

Comment:
could you add a comment to explain how the value has been computed?

Reply:
Ok, I will.